aten/src/ATen/native/vulkan/api/Runtime.cpp

```diff
@@ -274,6 +287,29 @@ std::unique_ptr<Runtime> init_global_vulkan_runtime() {
   return std::unique_ptr<Runtime>(nullptr);
 }
 
+// This is a utility function that can be used to trigger pre-emptive loading
+// of the global vulkan runtime in an asynchronous fashion. This is useful to
+// reduce the latency of the first loading of a vulkan model, as without calling
+// this function the first model load will trigger init_global_vulkan_runtime()
+// which will greatly increase model loading time.
+int async_runtime_loader() {
+  // This is safe since the global vulkan runtime instance is stored as a static
+  // local variable inside runtime(). Local static variable initialization is
+  // thread-safe since C++11.
+  std::thread async_loader(runtime);
```

Comment:
My worry is that you can still call `init_global_vulkan_runtime` from multiple threads and each of those can create an instance of Runtime. Only one of them will succeed in constructing `unique_ptr<Runtime>`. How can we validate this? 

However, you are still tying it to library load time so it probably is ok as you wont have instances where more than one thread is executing runtime.